docs/resources/dataarts_dataservice_catalog.md

```diff
@@ -0,0 +1,133 @@
+---
+subcategory: "DataArts Studio"
+layout: "huaweicloud"
+page_title: "HuaweiCloud: huaweicloud_dataarts_dataservice_catalog"
+description: |-
+  Use this resource to manage a shared or exclusive catalog within HuaweiCloud.
+---
+
+# huaweicloud_dataarts_dataservice_catalog
+
+Use this resource to manage a shared or exclusive catalog within HuaweiCloud.
+
+## Example Usage
+
+### Create a catalog under the ROOT path
+
+```hcl
+variable "workspace_id" {}
+variable "catalog_name" {}
+
+resource "huaweicloud_dataarts_dataservice_catalog" "test" {
+  workspace_id = var.workspace_id
+  dlm_type     = "EXCLUSIVE"
+  name         = var.catalog_name
+}
+```
+
+### Create a catalog under the specified catalog
+
+```hcl
+variable "workspace_id" {}
+variable "parent_catalog_id" {}
+variable "catalog_name" {}
+
+resource "huaweicloud_dataarts_dataservice_catalog" "test" {
+  workspace_id = var.workspace_id
+  parent_id    = var.parent_catalog_id
+  dlm_type     = "EXCLUSIVE"
+  name         = var.catalog_name
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region where the catalog is located.  
+  If omitted, the provider-level region will be used. Changing this parameter will create a new resource.
+
+* `workspace_id` - (Required, String, ForceNew) Specifies the ID of the workspace to which the catalog belongs.  
+  Changing this parameter will create a new resource.
+
+* `name` - (Required, String) Specifies the name of the catalog.  
+  The valid length is limited from `1` to `64`, only Chinese and English characters, digits and hyphens (-) are
+  allowed.  
+  The name must start with a Chinese or English character or a digit, and the Chinese characters must be in **UTF-8** or
+  **Unicode** format.
+
+* `description` - (Optional, String) Specifies the description of the catalog.  
+  Maximum of `255` characters are allowed.
+
+* `dlm_type` - (Optional, String, ForceNew) Specifies the type of DLM engine.  
+  The valid values are as follows:
+  + **SHARED**: Shared data service.
+  + **EXCLUSIVE**: The exclusive data service.
+
+  Defaults to **SHARED**. Changing this parameter will create a new resource.
+
+  -> The value of `dlm_type` for all catalogs under this directory must be consistent with this resource.
+
+* `parent_id` - (Optional, String) Specifies the ID of the parent catalog for current catalog.
+
+## Attribute Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The resource ID.
+
+* `path` - The path of current catalog.
+
+* `catalog_total` - The total number of sub-catalogs in the current catalog.
```

Comment:
If it is the number of subdirectories, is the naming of `catalog_total` not accurate enough.